sdk/src/main/scala/com/horizen/forge/Forger.scala

```diff
@@ -1,170 +1,138 @@
+  ExecutionContext.fromExecutor(Executors.newSingleThreadExecutor).execute(forgingInitiator)
 
-    val nextBockTimestamp = getTimeStampForEpochAndSlot(nextConsensusEpochNumber, nextConsensusSlotNumber)
-    val nextBlockEpochAndSlot = EpochAndSlot(nextConsensusEpochNumber, nextConsensusSlotNumber)
-    if(bestBlockEpochAndSlot > nextBlockEpochAndSlot) {
-      log.warn(s"Try to forge block with epochAndSlot ${nextBlockEpochAndSlot} but current best block epochAndSlot are: ${bestBlockEpochAndSlot}")
+  override def receive: Receive = {
+    processStartForgingMessage orElse processStopForgingMessage orElse processTryForgeNextBlockForEpochAndSlotMessage orElse {
+      case message: Any => log.error(s"Forger received strange message: ${message} from ${sender().path.name}")
     }
+  }
 
-    if ((nextConsensusEpochNumber - timeStampToEpochNumber(bestBlockInfo.timestamp)) > 1) log.warn("Forging is not possible: whole consensus epoch(s) are missed")
+  protected def processStartForgingMessage: Receive = {
+    case Forger.ReceivableMessages.StartForging => {
+      log.info("Receive StartForging message")
+      forgingIsActive = true
+      tryToCreateBlockNow()
+      sender() ! Success()
+    }
+  }
 
-    val consensusInfo: FullConsensusEpochInfo = view.history.getFullConsensusEpochInfoForNextBlock(bestBlockId, nextConsensusEpochNumber)
-    val totalStake = consensusInfo.stakeConsensusEpochInfo.totalStake
-    val vrfMessage = buildVrfMessage(nextConsensusSlotNumber, consensusInfo.nonceConsensusEpochInfo)
+  protected def processStopForgingMessage: Receive = {
+    case Forger.ReceivableMessages.StopForging => {
+      log.info("Receive StopForging message")
+      forgingIsActive = false
+      sender() ! Success()
+    }
+  }
 
-    val availableForgersDataWithSecret: Seq[ForgerDataWithSecrets] = view.vault.getForgingDataWithSecrets(nextConsensusEpochNumber).getOrElse(Seq())
+  protected def processTryForgeNextBlockForEpochAndSlotMessage: Receive = {
+    case TryForgeNextBlockForEpochAndSlot(epochNumber, slotNumber) => tryToCreateBlockForEpochAndSlot(epochNumber, slotNumber, Some(sender()))
+  }
 
-    val forgingDataOpt: Option[(ForgerDataWithSecrets, VRFProof)] = availableForgersDataWithSecret
-      .toStream
-      .map(forgerDataWithSecrets => (forgerDataWithSecrets, forgerDataWithSecrets.vrfSecret.prove(vrfMessage))) //get secrets thus filter forger boxes not owned by node
-      .find{case (forgerDataWithSecrets, vrfProof) => vrfProofCheckAgainstStake(forgerDataWithSecrets.forgerBox.value(), vrfProof, totalStake)} //check our forger boxes against stake
+  protected def tryToCreateBlockNow(): Unit = {
+    val currentTime: Long = Instant.now.getEpochSecond
+    val epochAndSlot = timestampToEpochAndSlot(currentTime)
+    log.info(s"Send TryForgeNextBlockForEpochAndSlot message with epoch and slot ${epochAndSlot}")
+    tryToCreateBlockForEpochAndSlot(epochAndSlot.epochNumber, epochAndSlot.slotNumber, None)
```

Comment:
If you don't want to put `Option[ActorRef]` as a parameter of `tryToCreateBlockForEpochAndSlot`, it can return `Try[ModifierId]` or even `ForgeResult` instead of `Unit`. So `tryToCreateBlockForEpochAndSlot` will not be responsible to respond external Actor if need.